SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestReplicationConditions/SpatialTestReplicationConditions.cpp

```diff
@@ -0,0 +1,604 @@
+			bCondIgnore[COND_OwnerOnly] = true;
+			bCondIgnore[COND_AutonomousOnly] = true;
+			bCondIgnore[COND_ReplayOrOwner] = true;
+			ProcessCommonActorProperties(bWrite, bCondIgnore);
+		}
+
+		if (AssertTrue(TestActor_CustomEnabled->AreAllDynamicComponentsValid(),
+					   TEXT("TestActor_CustomEnabled - All dynamic components should have arrived")))
+		{
+			const bool bWrite = false;
+			const bool bEnabled = true;
+			ProcessCustomActorProperties(TestActor_CustomEnabled, bWrite, bEnabled);
+		}
+
+		if (!bSpatialEnabled) // TODO: UNR-5212 - fix DOREPLIFETIME_ACTIVE_OVERRIDE replication
+		{
+			if (AssertTrue(TestActor_CustomDisabled->AreAllDynamicComponentsValid(),
+						   TEXT("TestActor_CustomDisabled - All dynamic components should have arrived")))
+			{
+				const bool bWrite = false;
+				const bool bEnabled = false;
+				ProcessCustomActorProperties(TestActor_CustomDisabled, bWrite, bEnabled);
+			}
+		}
+
+		if (AssertTrue(TestActor_AutonomousOnly->AreAllDynamicComponentsValid(),
+					   TEXT("TestActor_AutonomousOnly - All dynamic components should have arrived")))
+		{
+			const bool bWrite = false;
+			const bool bAutonomousExpected = false;
+			const bool bSimulatedExpected = true;
+			ProcessAutonomousOnlyActorProperties(bWrite, bAutonomousExpected, bSimulatedExpected);
+		}
+
+		if (AssertTrue(TestActor_PhysicsEnabled->AreAllDynamicComponentsValid(),
+					   TEXT("TestActor_PhysicsEnabled - All dynamic components should have arrived")))
+		{
+			const bool bWrite = false;
+			const bool bPhysicsEnabled = true;
+			const bool bPhysicsExpected = true;
+			ProcessPhysicsActorProperties(TestActor_PhysicsEnabled, bWrite, bPhysicsEnabled, bPhysicsExpected);
+		}
+
+		if (AssertTrue(TestActor_PhysicsDisabled->AreAllDynamicComponentsValid(),
+					   TEXT("TestActor_PhysicsDisabled - All dynamic components should have arrived")))
+		{
+			const bool bWrite = false;
+			const bool bPhysicsEnabled = false;
+			const bool bPhysicsExpected = true; // Gets replicated through simulated status
+			ProcessPhysicsActorProperties(TestActor_PhysicsDisabled, bWrite, bPhysicsEnabled, bPhysicsExpected);
+		}
+
+		FinishStep();
+	});
+
+	// Setup initial test variables
+	AddStep(TEXT("ASpatialTestReplicationConditions Initial Test Setup"), FWorkerDefinition::AllWorkers, nullptr, [this]() {
+
+		bSpatialEnabled = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();
+		ReplicationStage = STAGE_InitialReplication;
+		PropertyOffset = 0;
+
+		FinishStep();
+	});
+
+	// The Server spawns the TestActors and immediately after it creates and attaches the dynamic components
+	AddStep(TEXT("ASpatialTestReplicationConditions Spawn and Setup Test Actor"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AssertTrue(HasAuthority(), TEXT("Server 1 requires authority over the test actor"));
+
+		TestActor_Common = GetWorld()->SpawnActor<ATestReplicationConditionsActor_Common>(ActorSpawnPosition, FRotator::ZeroRotator,
```

Comment:
Wondering if we can do some sort of <UClass*, AActor*> array for these things we do for all these actors...
But may be overcomplicating it